services/hierarchy/src/main/kotlin/OrphanRemovalService.kt

```diff
@@ -86,220 +89,324 @@ class OrphanRemovalService(
         logger.info("Deleting orphaned children of ORT runs finished.")
     }
 
-    private suspend fun deleteOrphanedPackages() =
-        db.dbQuery {
-            PackagesTable.deleteWhere {
-                id notInSubQuery (
-                    PackagesAnalyzerRunsTable
-                        .select(PackagesAnalyzerRunsTable.packageId.alias("id"))
-                        .where(PackagesAnalyzerRunsTable.packageId.isNotNull())
-                    )
+    private suspend fun deleteOrphanedPackages(): Int {
+        var deletedRecords = 0
+        while (true) {
+            var recordsDeletedInIteration = 0
+            db.dbQuery {
+                PackagesTable.select(PackagesTable.id).where {
+                    PackagesTable.id notInSubQuery (
+                        PackagesAnalyzerRunsTable
+                            .select(PackagesAnalyzerRunsTable.packageId.alias("id"))
+                            .where(PackagesAnalyzerRunsTable.packageId.isNotNull())
+                        )
+                }.limit(DELETION_BATCH_SIZE).forEach {
+                    val packageId = it[PackagesTable.id]
+                    PackagesTable.deleteWhere { PackagesTable.id eq packageId }
```

Comment:
So actually you are not deleting batches of packages but individual packages now. Did you do any performance comparisons between before and after this change?